client/java/src/main/java/com/linecorp/centraldogma/client/AbstractWatcher.java

```diff
@@ -124,6 +150,13 @@ void start() {
         if (state.compareAndSet(State.INIT, State.STARTED)) {
             scheduleWatch(0);
         }
+        if (meterRegistry != null) {
+            // emit metrics once the values are ready
+            initialValueFuture.thenAccept(latest -> {
```

Comment:
Question) What do you think of just watching `this.latest.revision()` while setting -1 if not set?

I think the current approach introduces the additional complexity of maintaining an additional state.

e.g. Assuming:
```java
val watcher = dogma.watcher()...
watcher.start()
watcher.close()
// `initialValueFuture` is completed after the close and a gauge is registered
```

Reply:
I also struggled with this.  

Since I judged that it wouldn't become too complex, I wanted to provide as accurate metrics as possible, which is why I implemented it this way.  

However, if the **Central Dogma team** determines that it would make maintenance more complex, I think it's fine to set it to **-1**.

Reply:
Alternatively, `Double.NaN` could also be used

Reply:
If you prefer precise metrics, should we use a lock to register metrics, close the watcher and remove metrics? 
- The lock will solve the problem that jrhee17 mentioned.
- The metric registration could be placed in one method. 

```java
ReentrantLock lock = ...

if (initialValueFuture.complete(newLatest)) {
  updateWatchMetrics();
}
void updateWatchMetrics() {
   lock.lock();

   if (isStopped()) {
      return;
   }

   if (this.latest == null) {
     // register two metrics
   }
   // update the last received time

   lock.unlock();
}

void close() {
    lock.lock();

    ....
    lock.unlock();
}
```

Reply:
...Thinking about this topic again, the `-1` seems fine.

if we don't care about `-1`, we can simply write `... >= 0` in PromQL.

(I prefer `-1` because NaN has some quirks in handling.)